### PR TITLE
PBMC ingestion

### PR DESCRIPTION
**Pull request by jacoblurye**
Changes to allow manifest uploads accompanying https://github.com/CIMAC-CIDC/cidc-api-gae/pull/78

Now, clicking the "Upload" button actually triggers an upload. A successful upload produces a basic success message, and a failed upload just pastes the request error message to the "feedback" section in the manifest uploads card.

Also, adds a test for UI logic for the entire `TemplateUpload` component. As we add features throughout the UI, we'll want to add more tests like these (currently, there are nearly none).

### CODE REVIEW

File: src/api/api.ts [hunk 1]
```diff
@@ -93,19 +93,39 @@ function updateRole(
...
-        .post("ingestion/validate", formData, {
-            headers: { "content-type": "multipart/form" }
-        })
-        .then(res => _extractItem(res).errors);
+    return getApiClient(token).post(endpoint, formData, {
+        headers: { "content-type": "multipart/form" }
+    });
+}
+
+// TODO: determine the appropriate return type for this function based on API implementation.
```
Comment (curlup) [inline src/api/api.ts]: I guess you have that "API implementation" now =)
